### DISCUSSION TITLE
Tracking Cookies in Individual HTTP Requests with OpenWPM

### DISCUSSION THREAD

**Discussion by MohammadMahdiJavid** [openwpm/OpenWPM#1062] [Q&A] [answered]
Hi,

In general, when we visit a website, there are multiple HTTP requests made for various subresources, such as CSS files, JavaScript files, and images. these requests can contain cookies. My goal is to inspect the cookies sent with each of these individual HTTP requests, and I'm wondering if OpenWPM provides the functionality to achieve this.

I'd appreciate it if someone could guide me on how I can access and examine the cookies that are sent with each of these requests. If this is not currently supported, I'd also be interested in understanding if there are any potential workarounds or if this is something that could be considered for future development.

Thank you for your time and assistance.

**Comment by vringar** [comment, marked as answer]
Hey,

[as documented in MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies) cookies are sent via headers.
OpenWPM records the headers send as part of an HTTP Request [as documented here](https://openwpm.readthedocs.io/en/latest/Schema-Documentation.html#http-requests).
 
 You will need to extract the cookies from the headers, but that should just be splitting the headers string on newline and then checking with line begins with `Cookie:`
